crates/stackable-versioned-macros/src/attrs/item/mod.rs

```diff
@@ -0,0 +1,448 @@
+        }
+
+        // Now, iterate over all changes and ensure that their versions are
+        // between the added and deprecated version.
+        if !self.changes.iter().all(|r| {
+            added_version.map_or(true, |a| a < *r.since)
+                && deprecated_version.map_or(true, |d| d > *r.since)
+        }) {
+            return Err(Error::custom(
+                "all changes must use versions higher than `added` and lower than `deprecated`",
+            )
+            .with_span(item_ident));
+        }
+
+        Ok(())
+    }
+
+    /// This associated function is called by the top-level validation function
+    /// and validates that items use correct names depending on attached
+    /// actions.
+    ///
+    /// The following naming rules apply:
+    ///
+    /// - Fields or variants marked as deprecated need to include the
+    ///   deprecation prefix in their name. The prefix must not be included for
+    ///   fields or variants which are not deprecated.
+    fn validate_item_name(&self, item_ident: &impl ItemIdentExt) -> Result<()> {
+        let starts_with_deprecated = item_ident.starts_with_deprecated_prefix();
+
+        if self.deprecated.is_some() && !starts_with_deprecated {
+            return Err(Error::custom(format!(
+                "marked as `deprecated` and thus must include the `{deprecated_prefix}` prefix",
+                deprecated_prefix = item_ident.deprecated_prefix()
+            ))
+            .with_span(item_ident));
+        }
+
+        if self.deprecated.is_none() && starts_with_deprecated {
+            return Err(Error::custom(
+                format!("not marked as `deprecated` and thus mustn't include the `{deprecated_prefix}` prefix", deprecated_prefix = item_ident.deprecated_prefix())
+            ).with_span(item_ident));
+        }
+
+        Ok(())
+    }
+
+    fn validate_added_action(&self) -> Result<()> {
+        // NOTE (@Techassi): Can the path actually be empty?
+        if let Some(added) = &self.added {
+            if added.default_fn.segments.is_empty() {
+                return Err(Error::custom("`default_fn` cannot be empty")
+                    .with_span(&added.default_fn.span()));
+            }
+        }
+
+        Ok(())
+    }
+
+    /// This associated function is called by the top-level validation function
+    /// and validates that parameters provided to the `changed` actions are
+    /// valid.
+    fn validate_changed_action(&self, item_ident: &impl ItemIdentExt) -> Result<()> {
+        let mut errors = Error::accumulator();
+
+        // This ensures that `from_name` doesn't include the deprecation prefix.
+        for change in &self.changes {
+            if let Some(from_name) = change.from_name.as_ref() {
+                if from_name.starts_with(item_ident.deprecated_prefix()) {
+                    errors.push(
+                        Error::custom(
+                            "the previous name mustn't start with the deprecation prefix",
```

Comment:
I think the original line read better:

```suggestion
                            "the previous name must not start with the deprecation prefix",
```

(this will require an update to one of the stderr files)

Reply:
Adjusted in 14b868f